src/Prelude/Linear/Internal.hs

```diff
@@ -44,17 +44,17 @@ asTypeOf = const
 seq :: a -> b %1-> b
 seq x = Unsafe.toLinear (Prelude.seq x)
 
-($!) :: (a %1-> b) %1-> a %1-> b
+($!) :: (a %p-> b) %1-> a %p-> b
 ($!) f !a = f a
 
 -- | Beware, 'curry' is not compatible with the standard one because it is
 -- higher-order and we don't have multiplicity polymorphism yet.
-curry :: ((a, b) %1-> c) %1-> a %1-> b %1-> c
+curry :: ((a, b) %p-> c) %1-> a %p-> b %p-> c
 curry f x y = f (x, y)
 
 -- | Beware, 'uncurry' is not compatible with the standard one because it is
 -- higher-order and we don't have multiplicity polymorphism yet.
```

Comment:
Perhaps this comment needs an update now that the function uses multiplicity polymorphism.

Reply:
+1. You can generalize the remaining linear argument for `curry` and then remove the comment, it will completely subsume the one in Prelude. I've tried

```
($) :: (a %p-> b) %q-> a %p-> b
(&) :: a %p-> (a %p-> b) %q-> b
const :: a %m-> b -> a
asTypeOf :: a %m-> a -> a
($!) :: (a %p-> b) %q-> a %p-> b
curry :: ((a, b) %p-> c) %q-> a %p-> b %p-> c
uncurry :: (a %p-> b %p-> c) %q-> (a, b) %p-> c
forget :: (a %1-> b) %m-> a -> b
```

and the testsuite passed. I think it's reasonable to expect this to work at this complexity in 9.0. If there is a place where the monomorphic version works but the polymorphic doesn't, please let me know.